### DISCUSSION TITLE
Cannot load textures larger than 2048x2048

### DISCUSSION THREAD

**Discussion by Lukasz-pluszczewski** [replit/kaboom#714] [Q&A]
When trying to load a sprite larger than 2048 pixels, kaboom throws an error:
`texture size (1152 x 2640) exceeds limit (2048 x 2048)`

I have a lot of assets (especially tilemaps) that are way larger than that. Of course, I don't render the whole thing at once, but cutting assets into smaller chunks and recalculating where tiles are is a lot of work.

Is there a way to load those? Or is that limitation going to change in future versions?

**Comment by slmjkdbtl** [comment]
There's currently no way to load larger than 2048x2048. Generally it'll be better to render a tilemap by using one image for each tile or load tile images from a texture atlas, it's more flexible and take less memory (if your big tilemap image have lots of identical tiles it'll be wasting memory loading those identical pixels). What tool are you using to create the tilemap? It'll be great if it can export a file that specifies which tile is at which position and use that to add individual tiles in kaboom

**Reply by Lukasz-pluszczewski** [reply]
Thanks for the quick answer.
The problem is that I can't load texture atlas either (the image has the same limit). I have some files already created (from an art pack) and I create my levels using LDtk editor. It doesn't cut the source image into smaller ones and, to be honest, I don't want to do it either especially when prototyping and experimenting with different tiles. Seems like a waste of time when I just want to check different floors or walls out.

So to sum up:
- I have information about which tile is at which position and can use that either to render tiles manually or use it as a texture atlas, but the source images I have are larger than 2048px
- When I'm sure about the artwork I can create smaller files, sure, but when I'm experimenting I would love to be able to just load the whole thing to save time

The alternative (except manually creating smaller source files of course) is to do it programmatically: cut the images into smaller ones and recalculate coordinates from LDtk. Not sure which is less time-consuming :)

**Comment by csakip** [comment]
I'm also interested in this?
If you are not planning to implement it, could you suggest ideas how you would do it, and I can make a PR for it?

**Reply by slmjkdbtl** [reply]
I'm open to implementing it soon! There's a manual size limit because kaboom is using a fixed 2048x2048 texture atlas to batch user textures for faster rendering. I can make it just create a new texture and opt out of batch rendering if user loads a texture larger than 2048x2048.

**Comment by csakip** [comment]
Created a PR for this https://github.com/replit/kaboom/pull/833
